### PR TITLE
Move Hermes worker to make use of Worker Task Tracking

### PR DESCRIPTION
**Pull request by vanruch**

### CODE REVIEW

File: src/workers/hermes_worker.js [hunk 1]
```diff
@@ -10,22 +10,34 @@ This Source Code Form is “Incompatible With Secondary Licenses”, as defined
...
+    this.workId = null;
     if (!(this.strategy instanceof HermesUploadStrategy)) {
       throw new Error('A valid strategy must be provided');
     }
   }
 
   async periodicWork() {
-    await this.bundleCandidates();
-    await this.uploadWaitingCandidates();
+    this.workId = await this.workerTaskTrackingRepository.tryToBeginWork(HERMES_BUNDLING_WORK_NAME);
```
Comment (Anze1m) [inline src/workers/hermes_worker.js]: Why is this line not inside of any `try` block?

Comment (vanruch) [reply src/workers/hermes_worker.js]: It is, only in a parent class. When it fails, there's no point of removing finishWork

File: src/services/worker_task_tracking_repository.js [hunk 2]
```diff
@@ -21,7 +21,7 @@ export default class WorkerTaskTrackingRepository {
       if (error.message.includes(`duplicate key error`)) {
         throw new Error('Work of this type is currently in progress');
       }
-      throw new Error(error);
+      throw error;
```
Comment (spherefoundry) [inline src/services/worker_task_tracking_repository.js]: 👍

File: src/workers/hermes_worker.js [hunk 3]
```diff
@@ -13,20 +13,21 @@ import healthCheckHandler from '../routes/health_check';
 import PeriodicWorker from './periodic_worker';
 import HermesUploadStrategy from './hermes_strategies/upload_strategy';
 
+const HERMES_BUNDLING_WORK_NAME = 'HermesBundling';
```
Comment (spherefoundry) [inline src/workers/hermes_worker.js]: I think the proper term is "work type" isn't it?

File: src/workers/hermes_worker.js [hunk 4]
```diff
@@ -84,7 +90,11 @@ export default class HermesWorker extends PeriodicWorker {
   }
 
   async afterWorkLoop() {
-    await this.server.close();
-    await this.mongoClient.close();
+    try {
+      await this.workerTaskTrackingRepository.finishWork(this.workId);
```
Comment (spherefoundry) [inline src/workers/hermes_worker.js]: IMO: this is surplus. I wouldn't include the workId into the state of the worker. Instead I would introduce a local variable `workId` inside periodicWork, and fully handle it there.

File: src/workers/hermes_worker.js (hunk 1 above)
Comment (spherefoundry) [reply src/workers/hermes_worker.js]: It is a bad assumption that this is wrapped somewhere in a try block. All handling of the work should be done inside of this method.

**Pull request merged**